lib/matplotlib/axes/_base.py

```diff
@@ -2673,10 +2673,13 @@ def set_xscale(self, value, **kwargs):
         # around zero
         if value.lower() == 'log' and 'nonposx' not in kwargs.keys():
             kwargs['nonposx'] = 'clip'
-        self.xaxis._set_scale(value, **kwargs)
-        self.autoscale_view(scaley=False)
-        self._update_transScale()
-        self.stale = True
+
+        g = self.get_shared_x_axes()
+        for ax in g.get_siblings(self):
```

Comment:
Does get_siblings() return itself along with siblings? If not, then doesn't the `self` axes miss out on updating?